app/DynamicGraphs/GraphGenerator.hs

```diff
@@ -210,6 +200,11 @@ makeEdge id1 id2 = return $ DotEdge id1 id2 [ID (id1 `mappend` "|" `mappend` id2
 mappendTextWithCounter :: Text -> Integer -> Text
 mappendTextWithCounter text1 counter = text1 `mappend` "_counter_" `mappend` (pack (show counter))
 
+filterEmptyNodes :: [Tree [a]] -> [Tree [a]]
+filterEmptyNodes [] = []
```

Comment:
You can simplify this by using the built-in `filter` function. Note that `Tree` is an instance of `Eq`, so you can use `(==)` to compare `Tree`s.